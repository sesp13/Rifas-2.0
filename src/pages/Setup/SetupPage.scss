* {
  font-family: 'roboto';
}

.welcome-label {
  text-align: left;
  font-size: 15px;
}
