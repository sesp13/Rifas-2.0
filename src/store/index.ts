export * from './slices';
export * from './rootReducer';
export * from './store';