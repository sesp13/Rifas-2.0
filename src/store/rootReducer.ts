import { combineReducers } from '@reduxjs/toolkit';
import { gameSlice } from './slices';

export const rootReducer = combineReducers({
  game: gameSlice.reducer,
});

// Infer the `RootState` and `AppDispatch` types from the store itself
export type RootState = ReturnType<typeof rootReducer>;
