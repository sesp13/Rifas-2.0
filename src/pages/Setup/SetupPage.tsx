import { Button, Grid, TextField } from '@mui/material';
import { useForm } from '../../hooks/useForm';
import './SetupPage.scss';
import { FormEvent } from 'react';
import { useSelector, useDispatch } from 'react-redux';
import { RootState, setupGame } from '../../store';

export const SetupPage = () => {
  const gameState = useSelector((state: RootState) => state.game);
  const dispatch = useDispatch();

  const initialForm = {
    numberOfPlayers: 2,
    entryValue: gameState.entryValue,
    kickOutValue: gameState.kickOutValue,
    pointLimit: gameState.pointLimit,
  };

  const { onInputChange, numberOfPlayers, kickOutValue, entryValue, pointLimit } =
    useForm(initialForm);

  const submitSetupForm = (e: FormEvent) => {
    e.preventDefault();
    const action = setupGame({ entryValue, kickOutValue, pointLimit, numberOfPlayers });
    dispatch(action);
  };

  return (
    <>
      <form onSubmit={submitSetupForm}>
        <Grid container spacing={2}>
          <Grid item xs={12}>
            <p className="welcome-label">
              Bienvenido a la Rifa, por favor configura tu juego
            </p>
          </Grid>
          <Grid item xs={6}>
            <TextField
              fullWidth
              label="¿Cuántos jugadores?"
              name="numberOfPlayers"
              onChange={onInputChange}
              type="number"
              value={numberOfPlayers}
              variant="standard"
            />
          </Grid>
          <Grid item xs={6}>
            <TextField
              fullWidth
              label="Valor por volada"
              name="kickOutValue"
              onChange={onInputChange}
              type="number"
              value={kickOutValue}
              variant="standard"
            />
          </Grid>
          <Grid item xs={6}>
            <TextField
              fullWidth
              label="Valor de entrada inicial del juego"
              name="entryValue"
              onChange={onInputChange}
              type="number"
              value={entryValue}
              variant="standard"
            />
          </Grid>
          <Grid item xs={6}>
            <TextField
              fullWidth
              label="Tope de puntos"
              name="pointLimit"
              onChange={onInputChange}
              type="number"
              value={pointLimit}
              variant="standard"
            />
          </Grid>
          <Grid item xs={12}>
            <Button fullWidth variant="contained" type="submit">
              Enviar
            </Button>
          </Grid>
        </Grid>
      </form>
    </>
  );
};
