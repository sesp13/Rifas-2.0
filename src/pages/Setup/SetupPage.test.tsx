import { fireEvent, render, screen } from '@testing-library/react';
import { Provider } from 'react-redux';
import { BrowserRouter } from 'react-router-dom';
import { store } from '../../store';
import { SetupPage } from './SetupPage';

const mockedDispatch = jest.fn();

jest.mock('react-redux', () => ({
  ...jest.requireActual('react-redux'),
  useDispatch: () => mockedDispatch,
}));

const setupComponent = () =>
  render(
    <BrowserRouter>
      <Provider store={store}>
        <SetupPage />
      </Provider>
    </BrowserRouter>
  );

describe('Tests on <SetupPage />', () => {
  beforeEach(() => {
    jest.clearAllMocks();
  });

  test('should match the snapshot', () => {
    const { container } = setupComponent();
    expect(container).toMatchSnapshot();
  });

  test('should render setupForm', () => {
    const { container } = setupComponent();
    const form = container.querySelector('#setupForm');
    expect(form).toBeTruthy();
  });

  test('should dispatch an action when the form is submmited', () => {
    setupComponent();
    const submitBtn = screen.getByLabelText('submit-btn');
    fireEvent.click(submitBtn);
    expect(mockedDispatch).toHaveBeenCalled();
  });
});
