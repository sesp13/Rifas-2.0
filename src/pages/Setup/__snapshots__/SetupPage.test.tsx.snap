// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests on <SetupPage /> should match the snapshot 1`] = `
<div>
  <form
    id="setupForm"
  >
    <div
      class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2 css-mhc70k-MuiGrid-root"
    >
      <div
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 css-1idn90j-MuiGrid-root"
      >
        <p
          class="welcome-label"
        >
          Bienvenido a la Rifa, por favor configura tu juego
        </p>
      </div>
      <div
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6 css-1osj8n2-MuiGrid-root"
      >
        <div
          class="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root css-wb57ya-MuiFormControl-root-MuiTextField-root"
        >
          <label
            class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-sizeMedium MuiInputLabel-standard MuiFormLabel-colorPrimary MuiFormLabel-filled MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-sizeMedium MuiInputLabel-standard css-1c2i806-MuiFormLabel-root-MuiInputLabel-root"
            data-shrink="true"
            for=":r0:"
            id=":r0:-label"
          >
            ¿Cuántos jugadores?
          </label>
          <div
            class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl css-l4u8b9-MuiInputBase-root-MuiInput-root"
          >
            <input
              aria-invalid="false"
              class="MuiInputBase-input MuiInput-input css-1x51dt5-MuiInputBase-input-MuiInput-input"
              id=":r0:"
              name="numberOfPlayers"
              type="number"
              value="2"
            />
          </div>
        </div>
      </div>
      <div
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6 css-1osj8n2-MuiGrid-root"
      >
        <div
          class="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root css-wb57ya-MuiFormControl-root-MuiTextField-root"
        >
          <label
            class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-sizeMedium MuiInputLabel-standard MuiFormLabel-colorPrimary MuiFormLabel-filled MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-sizeMedium MuiInputLabel-standard css-1c2i806-MuiFormLabel-root-MuiInputLabel-root"
            data-shrink="true"
            for=":r1:"
            id=":r1:-label"
          >
            Valor por volada
          </label>
          <div
            class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl css-l4u8b9-MuiInputBase-root-MuiInput-root"
          >
            <input
              aria-invalid="false"
              class="MuiInputBase-input MuiInput-input css-1x51dt5-MuiInputBase-input-MuiInput-input"
              id=":r1:"
              name="kickOutValue"
              type="number"
              value="1000"
            />
          </div>
        </div>
      </div>
      <div
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6 css-1osj8n2-MuiGrid-root"
      >
        <div
          class="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root css-wb57ya-MuiFormControl-root-MuiTextField-root"
        >
          <label
            class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-sizeMedium MuiInputLabel-standard MuiFormLabel-colorPrimary MuiFormLabel-filled MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-sizeMedium MuiInputLabel-standard css-1c2i806-MuiFormLabel-root-MuiInputLabel-root"
            data-shrink="true"
            for=":r2:"
            id=":r2:-label"
          >
            Valor de entrada inicial del juego
          </label>
          <div
            class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl css-l4u8b9-MuiInputBase-root-MuiInput-root"
          >
            <input
              aria-invalid="false"
              class="MuiInputBase-input MuiInput-input css-1x51dt5-MuiInputBase-input-MuiInput-input"
              id=":r2:"
              name="entryValue"
              type="number"
              value="1000"
            />
          </div>
        </div>
      </div>
      <div
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6 css-1osj8n2-MuiGrid-root"
      >
        <div
          class="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root css-wb57ya-MuiFormControl-root-MuiTextField-root"
        >
          <label
            class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-sizeMedium MuiInputLabel-standard MuiFormLabel-colorPrimary MuiFormLabel-filled MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-sizeMedium MuiInputLabel-standard css-1c2i806-MuiFormLabel-root-MuiInputLabel-root"
            data-shrink="true"
            for=":r3:"
            id=":r3:-label"
          >
            Tope de puntos
          </label>
          <div
            class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl css-l4u8b9-MuiInputBase-root-MuiInput-root"
          >
            <input
              aria-invalid="false"
              class="MuiInputBase-input MuiInput-input css-1x51dt5-MuiInputBase-input-MuiInput-input"
              id=":r3:"
              name="pointLimit"
              type="number"
              value="100"
            />
          </div>
        </div>
      </div>
      <div
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 css-1idn90j-MuiGrid-root"
      >
        <button
          aria-label="submit-btn"
          class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-fullWidth MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-fullWidth css-1fu7jd5-MuiButtonBase-root-MuiButton-root"
          tabindex="0"
          type="submit"
        >
          Enviar
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </button>
      </div>
    </div>
  </form>
</div>
`;
